titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/validator/JobIamValidator.java

```diff
@@ -0,0 +1,133 @@
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.netflix.titus.runtime.endpoint.validator;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.titus.api.connector.cloud.IamConnector;
+import com.netflix.titus.api.iam.model.IamRole;
+import com.netflix.titus.api.iam.service.IamConnectorException;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.common.model.validator.EntityValidator;
+import com.netflix.titus.common.model.validator.ValidationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+/**
+ * This {@link EntityValidator} implementation validates and sanitizes Job IAM information.
+ */
+@Singleton
+public class JobIamValidator implements EntityValidator<JobDescriptor> {
+    private static final Logger logger = LoggerFactory.getLogger(JobImageValidator.class);
+    private static final String METRICS_ROOT = "titus.validation.iam";
+    private static final String failedMetricIamTag = "iamrole";
+    private static final String skippedMetricReasonTag = "reason";
+
+    private final JobSecurityValidatorConfiguration configuration;
+    private final IamConnector iamConnector;
+
+    private final Registry registry;
+    private final Id validationFailureId;
+    private final Id validationSkippedId;
+
+    @Inject
+    public JobIamValidator(JobSecurityValidatorConfiguration configuration, IamConnector iamConnector, Registry registry) {
+        this.configuration = configuration;
+        this.iamConnector = iamConnector;
+        this.registry = registry;
+        this.validationFailureId = registry.createId(METRICS_ROOT + ".failed");
+        this.validationSkippedId = registry.createId(METRICS_ROOT + ".skipped");
+    }
+
+    @Override
+    public Mono<Set<ValidationError>> validate(JobDescriptor jobDescriptor) {
+        if (isDisabled()) {
+            return Mono.just(Collections.emptySet());
+        }
+
+        String iamRoleName = jobDescriptor.getContainer().getSecurityProfile().getIamRole();
+
+        // Skip validation if no IAM was provided because a valid default will be used.
+        if (iamRoleName.isEmpty()) {
+            registry.counter(validationSkippedId.withTag(skippedMetricReasonTag, "noneProvided")).increment();
+            return Mono.just(Collections.emptySet());
+        }
+
+        // Skip any IAM that is not in "friendly" format. A non-friendly format is
```

Comment:
is this expected to come in the future, or is the plan to never validate these?

Reply:
Yes, expected to come in the future.